### DISCUSSION TITLE
[WASM][WindowsAppSDK][.NET6] Support for XamlRoot.RasterizationScale?

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#7775] [Q&A] [answered]
[XamlRoot.RasterizationScale](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.xamlroot.rasterizationscale?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(Microsoft.UI.Xaml.XamlRoot.RasterizationScale);k(DevLang-csharp)%26rd%3Dtrue&view=winui-3.0) appears not be supported. I would need that in my app for doing raw-pixels <-> view-pixels <-> points calculations.

Would there be a work around available or is there any other way to get that value (which e.g. on my machine is 1.25)?

Thanks

**Comment by jeromelaban** [comment]
You can do the same calculations manually using the APIs in `DisplayInformation`.

**Comment by DierkDroth** [comment]
You probably mean
`global::Windows.Graphics.Display.DisplayInformation.GetForCurrentView().RawPixelsPerViewPixel;`
... but GetForCurrentView() no longer works in WindowsAppSDK on desktop.

What else could be used? Or are you referring to something different?

**Reply by jeromelaban** [reply, marked as answer]
You can add some conditional code to use `DisplayInformation` on Uno, and `XamlRoot.RasterizationScale` on Windows until the property gets implemented.

**Comment by DierkDroth** [comment]
Got it. Thanks

**Comment by DierkDroth** [comment]
@jeromelaban I'm revisiting that issue and found that `XamlRoot.RasterizationScale` now is supported - at least for Uno WASM - and matches `global::Windows.Graphics.Display.DisplayInformation.GetForCurrentView().RawPixelsPerViewPixel` as well as `window.devicePixelRatio` in Javascript -> great. Thanks

(deleted some content, since not related to the case)
